metricbeat/helper/server/http/http.go

```diff
@@ -76,12 +83,21 @@ func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
 
 func (h *HttpServer) Start() error {
 	go func() {
-
-		logp.Info("Starting http server on %s", h.server.Addr)
-		err := h.server.ListenAndServe()
-		if err != nil && err != http.ErrServerClosed {
-			logp.Critical("Unable to start HTTP server due to error: %v", err)
+		if h.server.TLSConfig.Certificates != nil {
```

Comment:
Let's base this on whether or not `h.server.TLSConfig` is `nil`.

This may not be possible in this code path, but in general you can still use TLS while `Certificates` is empty if you provide a `GetCertificates()` method. That's how golang.org/x/crypto/acme/autocert works.

Reply:
I will do both checks so will check for nill and also check for the loaded certificate, just to assure that is an https server and not http. How about that? 

`if h.server.TLSConfig != nil && h.server.TLSConfig.Certificates != nil {`